src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProvider.cs

```diff
@@ -47,23 +62,102 @@ public ConsoleLoggerProvider(IOptionsMonitor<ConsoleLoggerOptions> options)
             }
         }
 
+        private static ConcurrentDictionary<string, IConsoleLogFormatter> SetFormatters(IEnumerable<IConsoleLogFormatter> fromSetup = null)
+        {
+            IEnumerable<IConsoleLogFormatter> formatters = fromSetup;
+            if (formatters == null || formatters.ToList().Count == 0)
+            {
+                var defaultMonitor = new FormatterOptionsMonitor<DefaultConsoleLogFormatterOptions>(new DefaultConsoleLogFormatterOptions());
+                var systemdMonitor = new FormatterOptionsMonitor<SystemdConsoleLogFormatterOptions>(new SystemdConsoleLogFormatterOptions());
+                formatters = new List<IConsoleLogFormatter>()
+                {
+                    new DefaultConsoleLogFormatter(defaultMonitor),
+                    new SystemdConsoleLogFormatter(systemdMonitor)
+                };
+            }
+            return new ConcurrentDictionary<string, IConsoleLogFormatter>(formatters.ToDictionary(f => f.Name));
+        }
+
+        // warning:  ReloadLoggerOptions can be called before the ctor completed,... before registering all of the state used in this method need to be initialized
         private void ReloadLoggerOptions(ConsoleLoggerOptions options)
         {
-            foreach (System.Collections.Generic.KeyValuePair<string, ConsoleLogger> logger in _loggers)
+            if (
+                options.FormatterName == null ||
+                !_formatters.TryGetValue(options.FormatterName, out IConsoleLogFormatter logFormatter) ||
+                !_formatters.TryGetValue(options.FormatterName?.ToLower(), out logFormatter)
```

Comment:
>ToLower [](start = 64, length = 7)

We need to be clear if we support casing in general. this one looks assume lowercasing be supported and not other casing be support. e.g. JSON for instance.

Reply:
It should be case insensitive